src/main/kotlin/fm/unit/UnitApp.kt

```diff
@@ -93,36 +139,31 @@ fun Application.module() {
         static("/static") {
             files("static")
         }
-        route("/reports") {
-            get("{key...}") {
-                // TODO(karsten): Lots of error handling and we don't support arbitrary prefix depth.
-                val key = call.parameters.getAll("key")?.joinToString("/") ?: ""
+        route("/reports/{organization}/{repository}/{prefix}") {
+
+            get {
+                val organization = call.parameters["organization"]
+                val repository = call.parameters["repository"]
+                val prefix = call.parameters["prefix"]
 
                 val reports = emptyList<Report>()// TODO(karsten): fetch from database
                 val summary = ProjectSummary(reports)
                 call.respondText(template(summary), ContentType.Text.Html)
             }
-            post("{key...}") {
-                val key = call.parameters.getAll("key")?.joinToString("/") ?: ""
+
+            post {
+                val organization = call.parameters["organization"]
+                val repository = call.parameters["repository"]
+                val prefix = call.parameters["prefix"] ?: ""
+
+                // TODO(karsten): Get ids by name.
```

Comment:
I'll do this in a follow up when I introduce the reads. We are also missing lots of error handling.